torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -0,0 +1,353 @@
+      view_func = [=](const at::Tensor& t) {
+        // We have to make another copy because the same lambda can be used twice! e.g., double backward
+        std::vector<c10::IValue> stack_args_copy_copy(stack_args_copy);
+        stack_args_copy_copy.at(stack_args_copy_copy.size() - num_arguments + aliased_input_idx) = t;
+        op.callBoxed(&stack_args_copy_copy);
+        return stack_args_copy_copy[stack_args_copy_copy.size() - num_returns + aliased_output_idx].toTensor();
+      };
+    }
+  }
+
+  {
+    at::AutoDispatchBelowADInplaceOrView guard;
+    op.redispatchBoxed(dispatch_keys & c10::after_ADInplaceOrView_keyset, stack);
+  }
+
+  if (any_is_inplace) {
+    for (const auto i : c10::irange(tensors_to_increment_version_on_stack.size())) {
+      increment_version(stack_args_copy[tensors_to_increment_version_on_stack[i]].toTensor());
+    }
+  }
+  // Differences between this boxed kernel and gen_inplace_or_view:
+  //  - we have an explicit list of view functions for which to generate the as_view logic
+  //    now that we rely on alias information. Can we remove this list since the info seems redundant?
+  //    - Probably not immediately (it would take some investigation).
+  //    - for multi-output views, there needs to be update to codegen parser (we would need to update
+  //      this regardless)
+  //      https://github.com/pytorch/pytorch/issues/47218
+  //    - many aliased functions aren't currently considered a view
+  //      however (probably because in these cases we'd want to return the original TensorImpl), all
+  //      autograd-explicit functions should be views if they have aliasing,
+  //      or there would be use_count issues.
+  //      TODO: we need an list of ops,e.g. _fw_primal, that are considered aliased for the sake
+  //      of jit schema, but don't have official view relationship
+  //  - What happens if view function returns non-tensor, non-tensorlist. Normally we'd raise an error
+  //    at codegen time if we detect that a view function returns.
+  //     - Should we have this check at run-time?
+  //       - Probably not. Extra run-time cost and "breaks" user code, but possibly
+  //         avoid maintaining extra codegen code
+  //  - Can make it a fallback? (We can do this in a separate PR. It would also simplify the UX)
+  //    - Probably yes, the main concern here is that users writing custom op must update their code
+  //      to use the correct guard. To help them, we we can raise a warning here if we detect
+  //      non-view ops going through this kernel.
+  //      - It should not result in an extra dispatch if AutoDispatchBelowADInplaceOrView guard is set
+  //        - it is for codegen'd ops (relies on list of view in gen_inplace_or_view?).
+  //        - for autogradNotImplemented custom ops, we also handle properly
+  //      - Inference mode already excludes the key, so no slowdown there.
```

Comment:
It excludes only the VariableType key.
But it doesn't exclude the InplaceOrView and so if a Tensor has that key, we will still end up in this iplementation.